### ISSUE TITLE
gnt-job list --filter and RAPI give wrong result with `not` filter

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1014]
Originally reported of Google Code with ID 958.

```
When using a `not` filter in `gnt-job --filter`, incorrect results are returned (same for the RAPI).
Example:

   $ gnt-job list
   4513 canceled TEST_DELAY(1.0)
   4514 canceled TEST_DELAY(1.0)
   4515 success  TEST_DELAY(1.0)

   $ gnt-job list --filter '(status == "success")'
   4515 success  TEST_DELAY(1.0)

   $ gnt-job list --filter 'not (status == "success")'
   [no output, should be 2 lines]

```
Originally added on 2014-09-25 13:28:38 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Found the reason, working on a fix.
```
Originally added on 2014-09-25 13:32:29 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed and pushed in 2117e8c.
```
Originally added on 2014-09-25 13:59:49 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-09-25 14:00:27 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-09-25 14:09:49 +0000 UTC.

Now blocking: 959 (now #1015);

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:37:03 +0000 UTC.

Changed State: Released
